### DISCUSSION TITLE
Ideas for Improving the Examples Section of all `admiral` Functions

### DISCUSSION THREAD

**Discussion by sophie-gem** [pharmaverse/admiral#2387] [General]
I've created this area as a place where anyone can 'brain dump' any potential ideas they have to improve the user experience when looking at the examples on the `admiral` website. Consider for example (I know it's super small - but hopefully it emphasises the point here), 

<img width="155" alt="image" src="https://github.com/pharmaverse/admiral/assets/90790226/ec02b2e6-23cc-4633-b327-be17822de73b">

This is the examples section for the [`derive_param_computed()`](https://pharmaverse.github.io/admiral/reference/derive_param_computed.html#ref-examples).

So if anyone has any ideas, has seen other R packages where it looks better, etc. please post here and we can decide on the best one! 🙂

**Comment by sophie-gem** [comment]
This comes from the discussion in https://github.com/pharmaverse/admiral/issues/2338. I will add the ideas already provided from there into this discussion thread tomorrow.

**Comment by sophie-gem** [comment]
@pharmaverse/admiral , @pharmaverse/admiral_comm - please feel free to add any ideas you have!

**Comment by bms63** [comment]
I love how the gt package is doing their examples, but looks like a lot of manual upkeep.  There has got to be a way to automate such tasks.

https://gt.rstudio.com/reference/tab_spanner.html

**Reply by ddsjoberg** [reply]
I think gt only has manual upkeep because the tables are rendered as images in a separate step, then the image is inserted after the code chunk.

But if we're just checking out the results that print to the console, we don't need to worry about that.  Here's what the examples section _could_ look like:

<img width="565" alt="image" src="https://github.com/pharmaverse/admiral/assets/26774684/751ba4ae-1458-483e-8eb8-7010a4edab50">

And that translate to this on the pkgdown site:

<img width="820" alt="image" src="https://github.com/pharmaverse/admiral/assets/26774684/873d7e95-920b-4e1b-8da9-378e50a68e21">

The BIG caveat to this approach is that the code will be run every. time. we. document. the. package, and that could really slow us down.

**Reply by sophie-gem** [reply]
Excuse if this is just my ignorance - but does there exist a way of 'caching' these r code chunks (a bit like you can with `knitr`) so it's not re-run each time? As that could potentially speed up the process? Definitely like the way `gt` does it though - makes it a lot clearer to read. Though perhaps a 'nice to have' (if it's possible), is to have subheadings under Examples (which go into the TOC) which splits it down further?

**Reply by bms63** [reply]
Doesn't the example code already run every time anyways?

**Reply by ddsjoberg** [reply]
it runs when we do the R CMD Check, but not every time we document (converting roxygen comments to rd syntax)

**Reply by bms63** [reply]
oh i see!! yes that could be a little cumbersome!!  

Perhaps we should do a canvas of all the functions where this improvement might have the most impact.  Some functions are pretty small and maybe don't need this update??  If only a few key functions need this update, then we can stress test and see how cumbersome it might be.

Other idea: Can we link out to an example vignette that is laid out nicer with more detailed examples?  Then keep the functions with minimal example and say - checkout the example vignette for a detailed walkthrough of each argument...

**Reply by manciniedoardo** [reply]
> Other idea: Can we link out to an example vignette that is laid out nicer with more detailed examples? Then keep the functions with minimal example and say - checkout the example vignette for a detailed walkthrough of each argument...

I'm not sure about this. I quite like that when you land on the reference page for a function, you have everything you need there on one page - arguments, examples, related functions, etc.

I'd rather do the canvas :)

**Comment by bms63** [comment]
@sophie-gem do you mind trying an experimenting to see how long the load time for a function with the examples re-rendered from a code chunk and examples without them rendered in the code chunk?

**Reply by sophie-gem** [reply]
Yes sure - will let you guys know the result!

**Comment by manciniedoardo** [comment]
Decision: this would be great to purse for maybe 1.3? We could establish a working group (leader tbc) and identify key functions to target for this "improved" reference page. Some could be `derive_vars_joined()`, `derive_param_computed()`, imputation functions.

**Comment by manciniedoardo** [comment]
How does this align with the roclets work done last year by @zdz2101 ?

**Comment by bms63** [comment]
@manciniedoardo and @bundfussr I think we can close this discussion as the Stefan's roclet solution has greatly improved our documentation layout and accessibility imo??WDYT?

**Reply by bundfussr** [reply]
Yes, I agree that we can close the issue.

**Reply by manciniedoardo** [reply]
agree
